core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -769,6 +769,8 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
 
       val killExecutors: Boolean => Future[Boolean] =
         if (executorsToKill.nonEmpty) {
+          executorsToKill.foreach(id =>
+            executorDataMap.get(id).foreach(_.executorEndpoint.send(StopExecutor)))
```

Comment:
Can we guarantee that `stop` is called before `kill` in this way?

Reply:
Yes.

Reply:
`StopExecutor` may arrive at executor after `kill` arrive at worker/container due to network delay, isn't it possible?